packages/@sanity/migrate/src/mutations/types.ts

```diff
@@ -16,45 +22,69 @@ export type SanityDocument = {
   _rev?: string
 }
 
+/**
+ * Represents a create-mutation that can be applied to a Sanity document.
+ */
 export type CreateMutation<Doc extends Optional<SanityDocument, '_id'>> = {
   type: 'create'
   document: Doc
 }
 
+/**
+ * Represents a createIfNotExists-mutation that can be applied to a Sanity document if it does not exist.
+ */
 export type CreateIfNotExistsMutation<Doc extends SanityDocument> = {
   type: 'createIfNotExists'
   document: Doc
 }
 
+/**
+ * Represents a createOrReplace mutation that can be applied to a Sanity document.
+ */
 export type CreateOrReplaceMutation<Doc extends SanityDocument> = {
   type: 'createOrReplace'
   document: Doc
 }
 
+/**
+ * Represents a delete mutation that can be applied to a Sanity document.
+ */
 export type DeleteMutation = {
   type: 'delete'
   id: string
 }
 
+/**
+ * Represents a patch mutation that can be applied to a Sanity document.
+ */
 export type PatchMutation<Patches extends NodePatchList = NodePatchList> = {
   type: 'patch'
   id: string
   patches: Patches
   options?: PatchOptions
 }
 
+/**
+ * Represents a mutation that can be applied to a Sanity document.
+ */
 export type Mutation<Doc extends SanityDocument = any> =
   | CreateMutation<Doc>
   | CreateIfNotExistsMutation<Doc>
   | CreateOrReplaceMutation<Doc>
   | DeleteMutation
   | PatchMutation
 
+/**
+ * A NodePatch represents a single operation that can be applied at a node at a specific path in a Sanity document.
+ */
 export type NodePatch<P extends Path = Path, O extends Operation = Operation> = {
   path: P
   op: O
 }
 
+/**
+ * Options for a patch operation.
+ */
 export type PatchOptions = {
   ifRevision?: string
```

Comment:
Can we add docs for `ifRevision` ? Not sure where to put that.

Reply:
Yep! Can probably refer to this section: https://www.sanity.io/docs/http-mutations#26600a871378

`patch(doc._id, [at('foo', set('bar'))], {ifRevision: doc._rev})`

Reply:
maybe? 
```suggestion
  /**
  * {@link https://www.sanity.io/docs/http-mutations#26600a871378}
  */
  ifRevision?: string
```